hazelcast/src/main/java/com/hazelcast/client/impl/clientside/HazelcastClientInstanceImpl.java

```diff
@@ -204,10 +205,15 @@ public HazelcastClientInstanceImpl(ClientConfig clientConfig,
         assert clientConfig != null || clientFailoverConfig != null : "At most one type of config can be provided";
         assert clientConfig == null || clientFailoverConfig == null : "At least one config should be provided ";
         if (clientConfig != null) {
+            MetricsConfigHelper.overrideClientMetricsConfig(clientConfig);
             this.config = clientConfig;
         } else {
+            for (ClientConfig failoverClientConfig : clientFailoverConfig.getClientConfigs()) {
+                MetricsConfigHelper.overrideClientMetricsConfig(failoverClientConfig);
+            }
             this.config = clientFailoverConfig.getClientConfigs().get(0);
         }
+        MetricsConfigHelper.overrideClientMetricsConfig(this.config);
```

Comment:
This line is not necessary. It is already overwritten in one of the paths above.